system/lib/wasmfs/emscripten.cpp

```diff
@@ -65,3 +67,41 @@ char *emscripten_get_preloaded_image_data_from_FILE(FILE *file,
   auto path = wasmfs::getPath(fd);
   return emscripten_get_preloaded_image_data(path.c_str(), w, h);
 }
+
+void emscripten_wget(const char* url, const char* file) {
+  // Create the ancestor directories.
+  auto segments = wasmfs::path::splitPath(file);
+  if (segments.empty()) {
+    return;
+  }
+  std::string currParent;
+  for (size_t i = 0; i < segments.size() - 1; i++) {
+    currParent += segments[i];
+    currParent += '/';
+    int result = mkdir(currParent.c_str(), 0777);
+    // Continue while we succeed in creating directories or while we see that
+    // they already exist.
+    if (result < 0 && errno != EEXIST) {
+      return;
+    }
+  }
+
+  // Fetch the data.
+  void* buffer;
+  int num;
+  int error;
+  emscripten_wget_data(url, &buffer, &num, &error);
+  if (error) {
+    return;
+  }
+
+  // Write the data.
+  FILE* f = fopen(file, "w");
+  if (f) {
+    fwrite(buffer, num, 1, f);
+    fclose(f);
```

Comment:
It might be more efficient to use `open` and `write` directly rather than stdio's buffered implementation, but then again, it might not make much difference. I guess there could be a code size difference if the user program doesn't use stdio?

Reply:
I agree.. using lower level syscalls would probably be a good idea here.. although any program calling this function is likely to then use the higher level FS calls itself.. so maybe not worth it?

Reply:
Sounds good, rewritten to use the lower-level calls. (Might not help much, but can't hurt I guess.)